test/core/codec/leb128/leb128_test.cpp

```diff
@@ -0,0 +1,127 @@
+                                       ::uint1024_t>;
+
+/**
+ * LEB128 encode samples
+ * @info produced with https://github.com/Equim-chan/leb128
+ */
+std::vector<std::pair<uint128_t, std::string>> leb128_samples{
+    {8, "08"},
+    {120, "78"},
+    {8888, "b845"},
+    {667888, "f0e128"},
+    {624485, "e58e26"},
+    {4815162342, "e69786f811"},
+    {2432615184, "9086fb8709"},
+    {1234567890, "d285d8cc04"},
+    {9876543210, "eaadc0e524"},
+    {19283746574839301, "85dca084f2cea022"},
+    {9876543212345678903u, "b7b8a3a593f7a1888901"},
+    {uint128_t("123456789012345678901234567890"),
+     "d295fcf1e49df8b9c3edbfc8ee31"}};
+
+TYPED_TEST_CASE(
+    LEB128CodecTester,
+    supported_types); /**< Initialize tests suite with possible cases */
+
+/**
+ * @given Min values of the supported numeric types
+ * @when Encoding/decoding value with LEB128 codec
+ * @then Decoded values must be the same as source
+ */
+TYPED_TEST(LEB128CodecTester, MinBoundariesIntegrity) {
+  ASSERT_TRUE(this->integrity(min_value<TypeParam>()));
+}
+
+/**
+ * @given Max values of the supported numeric types
+ * @when Encoding/decoding value with LEB128 codec
+ * @then Decoded values must be the same as source
+ */
+TYPED_TEST(LEB128CodecTester, MaxBoundariesIntegrity) {
+  ASSERT_TRUE(this->integrity(max_value<TypeParam>()));
+}
+
+/**
+ * @given LEB128 decoded/encoded samples
+ * @when Serialization/deserialization values with LEB128 codec
+ * @then Serializated values must be the same as samples
+ */
+TYPED_TEST(LEB128CodecTester, SerializeIntegrity) {
+  ASSERT_TRUE(this->serializeIntegrity(max_value<TypeParam>()));
+}
+
+/**
+ * @given Empty byte-vector to be decoded
+ * @when Decoding value with LEB128 codec
+ * @then Attempt to decode empty value must be failed
+ */
+TYPED_TEST(LEB128CodecTester, DecodeEmptyByteVector) {
+  std::vector<uint8_t> encoded{};
+  ASSERT_TRUE(this->checkDecodeFail(
+      encoded, static_cast<int>(LEB128DecodeError::INPUT_EMPTY)));
+}
+
+/**
+ * @given Byte-vector to be decoded, which is greater than decoded value
+ * @when Decoding value with LEB128 codec
+ * @then Attempt to decode too big value must be failed
+ */
+TYPED_TEST(LEB128CodecTester, DecodeTooBigByteVector) {
+  std::vector<uint8_t> encoded(0xFF, sizeof(TypeParam) + 1);
+  ASSERT_TRUE(this->checkDecodeFail(
+      encoded, static_cast<int>(LEB128DecodeError::INPUT_TOO_BIG)));
+}
+
+/**
+ * @given Hex-string with invalid value
+ * @when Deserializing hex-string with LEB128 codec
+ * @then Attempt to decode invalid hex-symbols must be failed
+ */
+TYPED_TEST(LEB128CodecTester, DeserializeInvalidValue) {
+  std::string invalid = "012abc!./";
+  ASSERT_TRUE(this->checkDeserializeFail(
+          invalid, static_cast<int>(LEB128DecodeError::INVALID_INPUT)));
+}
+
+/**
+ * @given LEB128-encoded samples
+ * @when Encoding value with LEB128 codec
+ * @then Encoded value must be the same as the sample
+ */
+TEST_P(LEB128AlgorithmTester, AlgorithmValidity) {
```

Comment:
FYC,
```suggestion
TEST_P(LEB128AlgorithmTester, SerializeSuccess) {
```
IMO, you don't need that checkSample method, you can do as follows:
```
  auto &&data = GetParam().first;
  auto &&sample = GetParam().second;
  ASSERT_EQ(sample, serialize(data));
```

Reply:
Serialize functions was removed